--- dynamic/msgregistry/message_registry.go	2019-06-06 20:53:22.444211200 +0300
+++ dynamic/msgregistry/message_registry.go	2019-06-06 20:51:52.610469400 +0300
@@ -147,7 +147,7 @@
 	for _, ed := range enums {
 		url := fmt.Sprintf("%s/%s", baseUrl, ed.GetFullyQualifiedName())
 		r.types[url] = ed
-		r.baseUrls[ed.GetFullyQualifiedName()] = url
+		r.baseUrls[ed.GetFullyQualifiedName()] = baseUrl
 	}
 }
 
@@ -155,7 +155,7 @@
 	for _, md := range msgs {
 		url := fmt.Sprintf("%s/%s", baseUrl, md.GetFullyQualifiedName())
 		r.types[url] = md
-		r.baseUrls[md.GetFullyQualifiedName()] = url
+		r.baseUrls[md.GetFullyQualifiedName()] = baseUrl
 		r.addEnumTypesLocked(baseUrl, md.GetNestedEnumTypes())
 		r.addMessageTypesLocked(baseUrl, md.GetNestedMessageTypes())
 	}
