diff --git i/BUILD.bazel w/BUILD.bazel
index fe131fb..f8522cc 100755
--- i/BUILD.bazel
+++ w/BUILD.bazel
@@ -68,6 +68,7 @@ par_binary(
     name = "pusher",
     srcs = ["tools/fast_pusher_.py"],
     main = "tools/fast_pusher_.py",
+    python_version = "PY2",
     visibility = ["//visibility:public"],
     deps = [":containerregistry"],
 )
@@ -76,6 +77,7 @@ par_binary(
     name = "digester",
     srcs = ["tools/image_digester_.py"],
     main = "tools/image_digester_.py",
+    python_version = "PY2",
     visibility = ["//visibility:public"],
     deps = [":containerregistry"],
 )
