diff --git i/BUILD.bazel w/BUILD.bazel
index b7b1f46..8fad37b 100755
--- i/BUILD.bazel
+++ w/BUILD.bazel
@@ -19,7 +19,9 @@ go_library(
         "@io_bazel_rules_go//go/platform:ios": [
             "@com_github_mitchellh_go_homedir//:go_default_library",
         ],
-        "//conditions:default": [],
+        "//conditions:default": [
+            "@com_github_mitchellh_go_homedir//:go_default_library",
+        ],
     }),
 )
 
