diff --git i/jsonpb/encode.go w/jsonpb/encode.go
index 685c80a..65d04ec 100755
--- i/jsonpb/encode.go
+++ w/jsonpb/encode.go
@@ -144,7 +144,12 @@ func (w *jsonWriter) marshalMessage(m protoreflect.Message, indent, typeURL stri
 				return fmt.Errorf("failed to marshal type URL %q to JSON: %v", typeURL, err)
 			}
 			js["@type"] = (*json.RawMessage)(&turl)
-			if b, err = json.Marshal(js); err != nil {
+			if indent != "" {
+				b, err = json.MarshalIndent(js, indent, indent)
+			} else {
+				b, err = json.Marshal(js)
+			}
+			if err != nil {
 				return err
 			}
 		}
